File: Ruby/CodeWars/8kyu-StringyStings.rb
=begin 
Write me a function stringy that takes a size and returns a string of alternating '1s' and '0s'.
the string should start with a 1.
a string with size 6 should return :'101010'.
with size 4 should return : '1010'.
with size 12 should return : '101010101010'.
The size will always be positive and will only use whole numbers.
=end

def stringy(size)
  (1..size).map{'10'}.join('').slice(0..size-1)
  (1..times).map{ |cont|  cont.odd? ? '1' : '0'}.join
end

puts stringy(0)

puts stringy(3)
#=> '101', 'Wrong result when size is 3')
puts stringy(5)
#=> '10101', 'Wrong result when size is 5')
puts stringy(12)
#=> '101010101010', 'Wrong result when size is 12')
puts stringy(26)
#=> '10101010101010101010101010', 'Wrong result when size is 26')
puts stringy(28)
#=> '1010101010101010101010101010', 'Wrong result when size is 28')

=begin ############################################# OTHERS SOLUTIONS 
2)
def stringy(size)
  (1..size).map { |i| i.even? ? '0' : '1' }.join('')
end
----------------------------------------------------------------------
3)
def stringy(size)
  "10" * (size / 2) + "1" * (size % 2)
end
----------------------------------------------------------------------
4)
def stringy(size)
  ('10'*size)[0,size]
end
----------------------------------------------------------------------
5)
def stringy(size)
  [1,0].cycle.take(size).join
end
----------------------------------------------------------------------
6)
def stringy(size)
  (1..size).map { |i| i % 2 }.join
end
----------------------------------------------------------------------
7)
def stringy(size)
  "10" * (size / 2) + "1" * (size % 2)
end

----------------------------------------------------------------------
8)
def stringy(size)
  ('10'*size)[0,size]
end
----------------------------------------------------------------------
9)
arr=[]
  size > 0 ? (0..size).map { |k| arr << '10'}.join('')[0..size-1] : ''

----------------------------------------------------------------------
10)

=end

=begin THRASH CAN ****************************************************
=end

File: JS/CodeWars/Storage/8kyu-StringyStings.js
/* 
Write me a function stringy that takes a size and returns a string of alternating '1s' and '0s'.
the string should start with a 1.
a string with size 6 should return :'101010'.
with size 4 should return : '1010'.
with size 12 should return : '101010101010'.
The size will always be positive and will only use whole numbers.
*/

function stringy(size) {
  let arr = [];
  for(let i=1;i<=size;i++){
    arr.push(i%2);
  }
  return arr.join('');
}
console.log(stringy(0));
console.log(stringy(3));
console.log(stringy(5));
console.log(stringy(12));
console.log(stringy(26));
console.log(stringy(28));
/* ############################################# OTHERS SOLUTIONS 
2)
const stringy = size => "10".repeat(size).slice(0,size);
----------------------------------------------------------------------
3)
const stringy = size => Array.from(Array(size), (v, i) => +!(i % 2)).join('');
----------------------------------------------------------------------
4)
const stringy = x => ''.padStart(x,'10');
----------------------------------------------------------------------
5)
function stringy(size) {
  return new Array(size).join('10').substr(0,size)   
}
----------------------------------------------------------------------
6)
function stringy(size) {
  return '10'.repeat(Math.floor(size / 2)) + (size % 2 ? '1' : '');
}
----------------------------------------------------------------------
7)
const stringy = (size) => String.repeat(10, (size + 1) >> 1).slice(0, size)
----------------------------------------------------------------------
8)

----------------------------------------------------------------------
9)

----------------------------------------------------------------------
10)

*/

/* THRASH CAN ****************************************************

*/